File: cpp_on_sea/monte_carlo_pi/typescript/montecarlopi.ts
// monte_carlo_pi.ts
// Calculates π using the Monte Carlo method with 10 million iterations

function estimatePi(iterations: number): number {
  let inside = 0;
  for (let i = 0; i < iterations; i++) {
    const x = Math.random();
    const y = Math.random();
    if (x * x + y * y <= 1) {
      inside++;
    }
  }
  return (inside / iterations) * 4;
}

const ITERATIONS = 10_000_000;
console.log(`Estimating π with ${ITERATIONS.toLocaleString()} iterations...`);
const piEstimate = estimatePi(ITERATIONS);
console.log(`Estimated π = ${piEstimate}`);

File: cpp_on_sea/monte_carlo_pi/ruby/montecarlopi.rb
# monte_carlo_pi.rb

# This Ruby script estimates the value of Pi using the Monte Carlo method
# with a default of 10 million iterations.

# Function to perform the Monte Carlo simulation
# iterations - number of random (x, y) points to generate
# Returns the estimated value of Pi

def estimate_pi(iterations)
  inside = 0

  iterations.times do
    x = rand
    y = rand
    inside += 1 if x * x + y * y <= 1.0
  end

  4.0 * inside / iterations
end

# Main execution block
if __FILE__ == $0
  # Default number of iterations: 10 million
  iterations = 10_000_000

  puts "Estimating Pi using Monte Carlo with \#{iterations} iterations..."

  pi_estimate = estimate_pi(iterations)

  puts "\nEstimated Pi: \#{pi_estimate}"
end
